stable-25-1-4, tsan: ydb/tests/olap/ttl_tiering/unstable_connection.py.TestUnstableConnection.test

https://github.com/ydb-platform/ydb/actions/runs/17081459148

https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run%20Small%20Medium/17081459148/ya-stable-25-1-4-x86-64-tsan/try_1/ya-test.html

ydb/tests/olap/ttl_tiering/unstable_connection.py.TestUnstableConnection.test 

```
teardown failed:
ydb/tests/olap/ttl_tiering/base.py:122: in teardown_class
    cls.cluster.stop()
ydb/tests/library/harness/kikimr_runner.py:577: in stop
    raise daemon.SeveralDaemonErrors(saved_exceptions)
E   ydb.tests.library.harness.daemon.SeveralDaemonErrors: Daemon failed with message: Bad exit_code..
E   Process exit_code = 100.
E   Stdout file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/olap/ttl_tiering/test-results/py3test/testing_out_stuff/unstable_connection.py.TestUnstableConnection.test/cluster/node_1/stdout
E   Stderr file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/olap/ttl_tiering/test-results/py3test/testing_out_stuff/unstable_connection.py.TestUnstableConnection.test/cluster/node_1/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   warning: address range table at offset 0x0 has a premature terminator entry at offset 0x10
E   warning: address range table at offset 0x30 has a premature terminator entry at offset 0x40
E   warning: address range table at offset 0x60 has a premature terminator entry at offset 0x70
E   warning: address range table at offset 0x90 has a premature terminator entry at offset 0xa0
E   warning: address range table at offset 0xc0 has a premature terminator entry at offset 0xd0
E   warning: address range table at offset 0x280 has a premature terminator entry at offset 0x290
E   warning: address range table at offset 0x2e0 has a premature terminator entry at offset 0x2f0
E   warning: address range table at offset 0x310 has a premature terminator entry at offset 0x320
E   warning: address range table at offset 0x340 has a premature terminator entry at offset 0x350
E   warning: address range table at offset 0x370 has a premature terminator entry at offset 0x380
E   warning: address range table at offset 0x3a0 has a premature terminator entry at offset 0x3b0
E   warning: address range table at offset 0x3d0 has a premature terminator entry at offset 0x3e0
E   warning: address range table at offset 0x400 has a premature terminator entry at offset 0x410
E   warning: address range table at offset 0x430 has a premature terminator entry at offset 0x440
E   warning: address range table at offset 0x460 has a premature terminator entry at offset 0x470
E   warning: address range table at offset 0x490 has a premature terminator entry at offset 0x4a0
E   warning: address range table at offset 0x4c0 has a premature terminator entry at offset 0x4d0
E   warning: address range table at offset 0x4f0 has a premature terminator entry at offset 0x500
E   warning: address range table at offset 0x520 has a premature terminator entry at offset 0x530
E   warning: address range table at offset 0x550 has a premature terminator entry at offset 0x560
E   warning: address range table at offset 0x580 has a premature terminator entry at offset 0x590
E   warning: address range table at offset 0x5b0 has a premature terminator entry at offset 0x5c0
E   warning: address range table at offset 0x5e0 has a premature terminator entry at offset 0x5f0
E   warning: address range table at offset 0x610 has a premature terminator entry at offset 0x620
E   warning: address range table at offset 0x640 has a premature terminator entry at offset 0x650
E   warning: address range table at offset 0xc40 has a premature terminator entry at offset 0xc50
E   warning: address range table at offset 0xc70 has a premature terminator entry at offset 0xc80
E   warning: address range table at offset 0xcd0 has a premature terminator entry at offset 0xce0
E   warning: address range table at offset 0xd00 has a premature terminator entry at offset 0xd10
E   warning: address range table at offset 0xd30 has a premature terminator entry at offset 0xd40
E   warning: address range table at offset 0xe20 has a premature terminator entry at offset 0xe30
E   warning: address range table at offset 0xe50 has a premature terminator entry at offset 0xe60
E   warning: address range table at offset 0xe80 has a premature terminator entry at offset 0xe90
E   warning: address range table at offset 0xeb0 has a premature terminator entry at offset 0xec0
E   warning: address range table at offset 0xee0 has a premature terminator entry at offset 0xef0
E   ==================
E   WARNING: ThreadSanitizer: data race (pid=451504)
E   Read of size 1 at 0x7214003317f8 by thread T62:
E   #0 OnExecute /-S/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h:82:14 (ydbd+0x27f6b599) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #1 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::Execute(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/common/script_cursor.cpp:17:13 (ydbd+0x27f6b599)
E   #2 NKikimr::NOlap::NReader::NCommon::TStepAction::DoExecuteImpl() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:30:33 (ydbd+0x27f65d04) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #3 NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8:19 (ydbd+0x27f7244a) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #4 NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/conveyor/usage/abstract.cpp:16:9 (ydbd+0x26f4dd0b) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #5 NKikimr::NConveyorComposite::TWorker::ExecuteTask(std::__y1::vector<NKikimr::NConveyorComposite::TWorkerTask, std::__y1::allocator<NKikimr::NConveyorComposite::TWorkerTask>>&&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:17:22 (ydbd+0x32635643) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #6 NKikimr::NConveyorComposite::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyorComposite::TEvInternal::TEvNewTask>, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.cpp:59:5 (ydbd+0x32638009) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #7 NKikimr::NConveyorComposite::TWorker::StateMain(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/tx/conveyor_composite/service/worker.h:39:13 (ydbd+0x3262b574) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #8 NActors::TActorCallbackBehaviour::Receive(NActors::IActor*, TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:272:9 (ydbd+0x18b06923) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #9 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.h:614:23 (ydbd+0x18b6db42) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #10 NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:269:28 (ydbd+0x18b69bcc) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #11 NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:460:39 (ydbd+0x18b6f6fa) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #12 NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:512:13 (ydbd+0x18b6f050) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #13 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:538:9 (ydbd+0x18b703f2) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #14 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:383:45 (ydbd+0x17772cb2) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   #15 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:244:20 (ydbd+0x1777318c) (BuildId: beb96070d0d4447759130acff763bf107e1f580e)
E   
E   Previous write of size 1 at 0x7214003317f8 by thread T61:
E   #0 OnE...

logsdir:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/olap/ttl_tiering/test-results/py3test/testing_out_stuff
log:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/olap/ttl_tiering/test-results/py3test/testing_out_stuff/unstable_connection.py.TestUnstableConnection.test.log
```

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
@@ -65,7 +65,7 @@ private:
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
     TAtomic StartInstant;
     TAtomic FinishInstant;
-    bool Started = false;
+    TAtomic Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)
@@ -79,9 +79,9 @@ public:
     }
 
     void OnExecute() {
-        if (!Started) {
+        if (!AtomicGet(Started)) {
             AtomicSet(StartInstant, TMonotonic::Now().MicroSeconds());
-            Started = true;
+            AtomicSet(Started, 1);
         }
     }
 

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
@@ -65,7 +65,7 @@ private:
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
     TAtomic StartInstant;
     TAtomic FinishInstant;
-    TAtomic Started = 0;
+    bool Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)
@@ -79,9 +79,8 @@ public:
     }
 
     void OnExecute() {
-        if (!AtomicGet(Started)) {
+        if (AtomicCas(&Started, 0, 1)) {
             AtomicSet(StartInstant, TMonotonic::Now().MicroSeconds());
-            AtomicSet(Started, 1);
         }
     }
 

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
@@ -65,7 +65,7 @@ private:
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
     TAtomic StartInstant;
     TAtomic FinishInstant;
-    bool Started = 0;
+    TAtomic Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/common/script.h
@@ -63,9 +63,8 @@ class TFetchingScript {
 private:
     YDB_READONLY_DEF(TString, BranchName);
     std::vector<std::shared_ptr<IFetchingStep>> Steps;
-    TAtomic StartInstant;
+    TAtomic StartInstant = 0;
     TAtomic FinishInstant;
-    TAtomic Started = 0;
 
 public:
     TFetchingScript(const TString& branchName, std::vector<std::shared_ptr<IFetchingStep>>&& steps)
@@ -79,9 +78,7 @@ public:
     }
 
     void OnExecute() {
-        if (AtomicCas(&Started, 0, 1)) {
-            AtomicSet(StartInstant, TMonotonic::Now().MicroSeconds());
-        }
+        AtomicCas(&StartInstant, TMonotonic::Now().MicroSeconds(), 0);
     }
 
     TString DebugString() const;

